### ISSUE TITLE
BeanPropertyMap NPE when enable ACCEPT_CASE_INSENSITIVE_PROPERTIES. version: 2.11.0.rc1

### ISSUE THREAD

**Issue by chowtin** [FasterXML/jackson-databind#2669]
```
    public BeanPropertyMap(boolean caseInsensitive, Collection<SettableBeanProperty> props,
            Map<String,List<PropertyName>> aliasDefs,
            Locale locale)
    {
        _caseInsensitive = caseInsensitive;
        _propsInOrder = props.toArray(new SettableBeanProperty[props.size()]);
        _aliasDefs = aliasDefs;
        _aliasMapping = _buildAliasMapping(aliasDefs);
        _locale = locale;
        init(props);
    }

....

    private Map<String,String> _buildAliasMapping(Map<String,List<PropertyName>> defs)
    {
        if ((defs == null) || defs.isEmpty()) {
            return Collections.emptyMap();
        }
        Map<String,String> aliases = new HashMap<>();
        for (Map.Entry<String,List<PropertyName>> entry : defs.entrySet()) {
            String key = entry.getKey();
            if (_caseInsensitive) {
                key = key.toLowerCase(_locale);
            }
            for (PropertyName pn : entry.getValue()) {
                String mapped = pn.getSimpleName();
                if (_caseInsensitive) {
                    mapped = mapped.toLowerCase();
                }
                aliases.put(mapped, key);
            }
        }
        return aliases;
    }
```

Issue is from the order of below two statements.
```
        _aliasMapping = _buildAliasMapping(aliasDefs);
        _locale = locale;
```
BeanPropertyMap::_buildAliasMapping used a un-initialized ```_locale```.

stacktrace:
```
java.lang.NullPointerException
	at java.lang.String.toLowerCase(String.java:2563)
	at com.fasterxml.jackson.databind.deser.impl.BeanPropertyMap._buildAliasMapping(BeanPropertyMap.java:820)
	at com.fasterxml.jackson.databind.deser.impl.BeanPropertyMap.<init>(BeanPropertyMap.java:100)
```

**Comment by cowtowncoder** [comment]
Thank you for reporting the issue.

One thing that I'd need, in addition to useful information already included, would be example code that triggers the problem. This because none of existing unit tests hits that so I think there  is some specific configuration or usage that causes this.

**Comment by chowtin** [comment]
below is the testcase. 
add ```JsonAlias``` on one field
```
    static class Pojo {

        @JsonAlias({"nick", "name"})
        private String name;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    @Test
    public void testJackson() throws JsonProcessingException {

        ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);

        String text = "{\"name\":\"test\"}";
        Pojo pojo = objectMapper.readValue(text, Pojo.class);
        Assert.assertNotNull(pojo);
    }

```

**Comment by cowtowncoder** [comment]
@chowtin thank you!

**Comment by cowtowncoder** [comment]
@chowtin I can not reproduce the issue: which version are you using? Would it be possible to verify with the latest released version (2.10.3)?

**Comment by chowtin** [comment]
2.10.3 works.

2.11.0.rc1 raise error.  (I should use a released version)

**Comment by cowtowncoder** [comment]
@chowtin Hmmmh. Ok, thank you for verifying. I think what happened here was that I tested with 2.10 branch (since if it was failing there, I would want to fix it for 2.10.4 too). But did not yet merge test into 2.11.

Now I can reproduce it with 2.11 branch. Thank you!

**Comment by cowtowncoder** [comment]
Ok good -- seconds rc1-only bug found, now fixed. Thank you @chowtin for the report!
